### PR TITLE
[spark] Support spark DropPartition with sync metastore

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose

Linked issue: close #xxx
Currently after paimon spark drop partition not sync to metastore like flink(https://github.com/apache/paimon/pull/3411), this pr is aim to support it.

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala [hunk 1]
```diff
@@ -69,10 +69,21 @@ trait PaimonPartitionManagement extends SupportsAtomicPartitionManagement {
...
+        var metastoreClient: MetastoreClient = null
+        val clientFactory = fileStoreTable.catalogEnvironment().metastoreClientFactory
         try {
           commit.dropPartitions(partitions.toSeq.asJava, BatchWriteBuilder.COMMIT_IDENTIFIER)
+          // syn to metastore with delete partitions
+          if (clientFactory != null) {
+            metastoreClient = clientFactory.create()
+            toPaimonPartitions(rows).foreach(metastoreClient.deletePartition)
+          }
+
```
Comment (wwj6591812) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala:81]: delete excess blank line

**Comment by xuzifu666** [comment]
@wwj6591812 Thanks, had addressed

File: paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala [hunk 2]
```diff
@@ -69,10 +69,20 @@ trait PaimonPartitionManagement extends SupportsAtomicPartitionManagement {
       case fileStoreTable: FileStoreTable =>
         val partitions = toPaimonPartitions(rows).map(_.asInstanceOf[JMap[String, String]])
         val commit: FileStoreCommit = fileStoreTable.store.newCommit(UUID.randomUUID.toString)
+        var metastoreClient: MetastoreClient = null
+        val clientFactory = fileStoreTable.catalogEnvironment().metastoreClientFactory
         try {
           commit.dropPartitions(partitions.toSeq.asJava, BatchWriteBuilder.COMMIT_IDENTIFIER)
+          // sync to metastore with delete partitions
+          if (clientFactory != null) {
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala:77]: should check `metastore.partitioned-table`.

Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala:77]: Also for `createPartitions`.

Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala]: Yes，had added a check for metastore.partitioned-table and case. Thanks @JingsongLi

Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/PaimonPartitionManagement.scala]: Done

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**